stories/__tests__/SQFormRadioButtonGroup.stories.test.js

```diff
@@ -0,0 +1,173 @@
+
+        expect(firstRadioButton).not.toBeChecked();
+        userEvent.click(firstRadioButton);
+        expect(firstRadioButton).toBeChecked();
+      });
+
+      it('allows selection via label', () => {
+        render(<SQFormRadioButtonGroup size="auto" />);
+
+        const giantPandaRadioButton = screen.getAllByRole('radio')[0];
+        const giantPandaLabel = screen.getByText(/Giant Panda/i);
+
+        expect(giantPandaRadioButton).not.toBeChecked();
+        userEvent.click(giantPandaLabel);
+        expect(giantPandaRadioButton).toBeChecked();
+      });
+
+      it('allows only one value to be selected', () => {
+        render(<SQFormRadioButtonGroup size="auto" />);
+
+        const radioButtons = screen.getAllByRole('radio');
+        const firstRadioButton = radioButtons[0];
+        const secondRadioButton = radioButtons[1];
+
+        expect(firstRadioButton).not.toBeChecked();
+        expect(secondRadioButton).not.toBeChecked();
+
+        userEvent.click(firstRadioButton);
+        expect(firstRadioButton).toBeChecked();
+        expect(secondRadioButton).not.toBeChecked();
+
+        userEvent.click(secondRadioButton);
+        expect(firstRadioButton).not.toBeChecked();
+        expect(secondRadioButton).toBeChecked();
+      });
+    });
+
+    describe('onChange handler', () => {
+      let onChangeMock;
+
+      beforeEach(() => {
+        onChangeMock = jest.fn();
+      });
+
+      it('calls onChange handler on a change', () => {
+        render(<SQFormRadioButtonGroup size="auto" onChange={onChangeMock} />);
+
+        const firstRadioButton = screen.getAllByRole('radio')[0];
+        userEvent.click(firstRadioButton);
+        expect(onChangeMock).toHaveBeenCalledTimes(1);
+      });
+    });
+  });
+
+  it('Has default value with an initialValue', () => {
+    const initialValues = {
+      pandas: 'red panda'
+    };
+    render(
+      <SQFormRadioButtonGroup size="auto" SQFormProps={{initialValues}} />
+    );
+
+    const radioButtons = screen.getAllByRole('radio');
+
+    const firstRadioButton = radioButtons[0];
+    const secondRadioButton = radioButtons[1];
+    const thirdRadioButton = radioButtons[2];
+
+    expect(firstRadioButton).not.toBeChecked();
+    expect(secondRadioButton).toBeChecked();
+    expect(thirdRadioButton).not.toBeChecked();
+  });
+
+  describe('With validation', () => {
+    describe('Not touched', () => {
+      it('has submit button disabled', () => {
+        render(<SQFormRadioButtonGroupWithValidation size="auto" />);
+
+        const submitButton = screen.getByRole('button', {
+          name: /form submission/i
+        });
+
+        expect(submitButton).toBeDisabled();
+      });
+
+      it('displays required helper text', () => {
+        render(<SQFormRadioButtonGroupWithValidation size="auto" />);
+
+        const requiredText = screen.getByText('Required');
+        expect(requiredText).toBeInTheDocument();
+        expect(requiredText).toHaveClass('Mui-required');
```

Comment:
A test for checking that the required text gets the `Mui-error` class when appropriate would be a good addition as well. That could probably be achieved by tabbing over the input with `userEvent`, similar to some of the component tests